rpcserver.go

```diff
@@ -6745,6 +6751,222 @@ func (r *rpcServer) ListPermissions(_ context.Context,
 	}, nil
 }
 
+func (r *rpcServer) updateFeatures(updates []*lnrpc.UpdateFeatureAction) (*lnrpc.Op, error) {
+
+	ops := &lnrpc.Op{Entity: "features"}
+
+	fsets := make(map[feature.Set]*lnwire.RawFeatureVector)
+	sets := []feature.Set{feature.SetNodeAnn}
+
+	for _, set := range sets {
+		raw := r.server.featureMgr.GetRaw(set)
+
+		for _, update := range updates {
+			bit := lnwire.FeatureBit(update.FeatureBit)
+			switch update.Action {
+			case lnrpc.UpdateAction_ADD:
+				if !raw.IsSet(bit) && raw.SafeSet(bit) != nil {
+					raw.Set(bit)
+					ops.Actions = append(ops.Actions,
+						fmt.Sprintf("%s set in %s",
+							lnwire.Features[bit],
+							set.String()))
+				}
+			case lnrpc.UpdateAction_REMOVE:
+				if raw.IsSet(bit) {
+					raw.Unset(bit)
+					ops.Actions = append(ops.Actions,
+						fmt.Sprintf("%s unset in %s",
+							lnwire.Features[bit],
+							set.String()))
+				}
+			default:
+				return nil, fmt.Errorf("invalid feature update action: %v", update.Action)
+			}
+		}
+
+		// Ensure that all of our feature sets properly set any
+		// dependent features.
+		fv := lnwire.NewFeatureVector(raw, lnwire.Features)
+		err := feature.ValidateDeps(fv)
```

Comment:
Do we need to validate anything else?

Reply:
See above comment, this ensures the features are well defined, but can possibly allow a user to _over write_ a feature we set (like anchor channels), which we want to prevent IMO.